src_code/src/controllers/services/Inventory/GetAvailableModels.js

```diff
@@ -0,0 +1,102 @@
+/** Initialize neccessary modules */
+const responseBuilder = require("../../../utils/interfaces/IResponseBuilder");
+const db = require("../../../configurations/database/DatabaseConfigurations");
+const dbHelpers = require("../../../utils/interfaces/IDBHelperFunctions");
+
+/**
+ * Retrieve available models within a specified date range.
+ * @param {Object} res - Express response object.
+ * @param {string} startDate - Start date for availability check.
+ * @param {string} endDate - End date for availability check.
+ * @returns {Object} - Response object indicating success or failure.
+ */
+async function GetAvailableModels(res, startDate, endDate) {
+    try{
+        /** Validate the date period */
+        const datePeriodError = DatePeriodValidation(startDate, endDate);
+        if(datePeriodError) {
+            return responseBuilder.BadRequest(res, datePeriodError);
+        }
+
+        /** Fetch all models from the database */
+        const allModels = await db("equipment_model")
+        .select(
+            "equipment_type.TYPE_NAME as typeName",
+            "equipment_model.FK_TYPE_ID AS typeId",
+            "equipment_model.PK_MODEL_ID as modelId",
+            "equipment_model.MODEL_NAME as modelName",
+            "equipment_model.MODEL_PHOTO_URL as modelPhoto"
+        )
+        .join("equipment_type", "equipment_type.PK_TYPE_ID", "=", "equipment_model.FK_TYPE_ID");
+
+    /** Check if there are no models, return a BadRequest response */
+    if (!allModels || allModels?.length === 0) {
+        return responseBuilder.BadRequest(res, "There is no model available in this period of time.");
```

Comment:
Is there a reason why this is also a BadRequest?

Reply:
See comments above

Reply:
We have been doing this logic for all APIs in the codebase.